pkg/prometheus/promcfg.go

```diff
@@ -2142,6 +2154,39 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 	}
 }
 
+func generateRoleSelectorConfig(role *monitoringv1alpha1.KubernetesRole, selector *metav1.LabelSelector, k8sSDConfig yaml.MapSlice, cg *ConfigGenerator) yaml.MapSlice {
```

Comment:
(nit) can we have `k8sSDConfig yaml.MapSlice` as the first arg (to mimic the append() function)?